pkg/agent/handler/credential.go

```diff
@@ -356,6 +356,8 @@ func (h *credentials) provisionPostProcess(status prov.RequestStatus, credential
 			cred.Spec.State.Rotate = false
 			h.client.UpdateResourceInstance(cred)
 		}
+	} else if cred.State.Name == "" {
```

Comment:
So setting State.Name to v1.Inactive only happens if status level is not success?

Reply:
Not success and no existing state, otherwise the state of that cred stays as is.

Reply:
It was getting set for a success. Now it gets set when there is an error too.